avocado/core/loader.py

```diff
@@ -691,8 +690,7 @@ def ignore_broken(klass, uid, params=None):
                                                 subtests_filter)
             else:
                 if os.access(test_path, os.X_OK):
-                    return self._make_test(test.SimpleTest,
-                                           pipes.quote(test_path))
+                    return self._make_test(test.SimpleTest, test_path)
```

Comment:
OK, the system of execution probably changed since then. I played with your version and it works well. To be sure not to break things I'd suggest updating the simple check selftest with ` ` (space) in the script name and at least one argument and execute it with the argument (eg. `[ "$1" != "correct_arg" ] && exit -1`). That should increase the test coverage.